easybuild/easyblocks/c/cppcheck.py

```diff
@@ -0,0 +1,139 @@
+        """
+        if self.cfg['build_gui']:
+            os.chdir(self.cfg['start_dir']+'/gui')
+            if self.cfg['have_rules']:
+                cmd = 'qmake HAVE_RULES=yes'
+            else:
+                cmd = 'qmake'
+            run_cmd(cmd, log_all=True, simple=True)
+            os.chdir(self.cfg['start_dir'])
+        else:
+            pass
+
+    def build_step(self, verbose=False):
+        """
+        Compile cppcheck with make and cppcheck-gui with qmake and make 
+        """
+        self.cfg.update('buildopts', 'CFGDIR=%(installdir)s/cfg')
+        
+        if self.cfg['have_rules']:
+            self.cfg.update('buildopts', 'HAVE_RULES=yes')
+
+        super(EB_cppcheck, self).build_step(verbose=verbose)
+        
+        if self.cfg['build_gui']:
+            super(EB_cppcheck, self).build_step(verbose=verbose,path='gui')
+
+    def install_step(self):
+        """
+        Install cppcheck with make install and cppcheck-gui copying the file
+        """
+        self.cfg.update('installopts', 'DESTDIR=%(installdir)s/ PREFIX="" CFGDIR=cfg')
+        
+        super(EB_cppcheck, self).install_step()
+       
+        # The GUI doesn't have make install, so we copy it manually
+        if self.cfg['build_gui']:
+            try:
+                # make sure we're (still) in the start dir
+                os.chdir(self.cfg['start_dir'])
+
+                file_to_copy = 'cppcheck-gui'
+                self.log.debug("Copying cppcheck-gui")
+
+                filepath = self.cfg['start_dir'] + '/gui/' + file_to_copy
+
+                # Perform the copy
+                if os.path.isfile(filepath):
+                    target_dest = os.path.join(self.installdir, 'bin')
+                    self.log.debug("Copying file %s to %s" % (filepath, target_dest))
+                    shutil.copy2(filepath, target_dest)
+                else:
+                    raise EasyBuildError("Can't copy non-existing path %s to %s", filepath, target)
+                
+                # Create cfg symlink. It shouldn't be necessary, but cppcheck-gui complaints otherwise
+                # unless called with --data-dir
+                src = os.path.join(self.installdir, 'cfg')
+                dst = os.path.join(self.installdir, 'bin/cfg')
+                self.log.debug("Creating symlink %s to %s" % (src, dst))
+                os.symlink(src, dst)
+
+            except OSError, err:
+                raise EasyBuildError("Copying %s to installation dir failed: %s", fil, err)
+
+    def sanity_check_step(self):
+        """
+        Custom sanity check for cppcheck.
+        """
+
+        if self.cfg['build_gui']:
+            custom_paths = {
+                'files': ['bin/cppcheck', 'bin/cppcheck-gui' ],
+                'dirs': ['.', 'bin/cfg']
+            }
+        else:
+            custom_paths = {
+                'files': ['bin/cppcheck' ],
+                'dirs': ['.']
+            }
```

Comment:
@damianam I would make it more explicit what is different in case of `build_gui` (and don't use `.` as an entry):

```python
custom_paths = {
    'files': ['bin/cppcheck'],
    'dirs': [],
}
if self.cfg['build_gui']:
    custom_paths['files'].append('bin/cppcheck-gui')
    custom_paths['dirs'].append('bin/cfg')
```